src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -433,8 +439,17 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
 #if DEBUG // Can remove ifdefs and replace with Conditional after https://github.com/dotnet/roslyn/issues/47463 is fixed
                 static void verifyDataConsistent(CommonMethodEarlyWellKnownAttributeData? earlyData, MethodWellKnownAttributeData? lateData)
                 {
-                    Debug.Assert((earlyData, lateData) is ((null or { UnmanagedCallersOnlyAttributePresent: false }), (null or { UnmanagedCallersOnlyAttributeData: null }))
-                                                          or ({ UnmanagedCallersOnlyAttributePresent: true }, { UnmanagedCallersOnlyAttributeData: not null }));
+                    if (lateData is { UnmanagedCallersOnlyAttributeData: not null })
+                    {
+                        // We can't verify the symmetric case here. Error conditions (such as if a bad expression was provided to the array initializer)
+                        // can cause the attribute to be entirely unbound during regular attribute binding. Early binding doesn't know that though, so
```

Comment:
>attribute to be entirely unbound [](start = 41, length = 32)

"to be unbound" sounds like binding is done, then undone. Perhaps "can leave the attribute unbound" or "can cause the attribute to be skipped".